Storage/SQL/SQLiteHistory.swift

```diff
@@ -323,10 +323,12 @@ extension SQLiteHistory: BrowserHistory {
         let whereClause: String
         let whereFragment = (whereData == nil) ? "" : " AND (\(whereData!))"
         if let filter = filter {
-            args = ["%\(filter)%", "%\(filter)%"]
-
+            // Split filter on whitespace and set up where clause for every word.
+            let words = filter.componentsSeparatedByCharactersInSet(NSCharacterSet.whitespaceCharacterSet()).filter { !$0.isEmpty }
             // No deleted item has a URL, so there is no need to explicitly add that here.
-            whereClause = " WHERE ((\(TableHistory).url LIKE ?) OR (\(TableHistory).title LIKE ?)) \(whereFragment)"
+            let filterFragment = "(" + " OR ".join(words.map { _ in "(\(TableHistory).url LIKE ?) OR (\(TableHistory).title LIKE ?)" }) + ")"
```

Comment:
I think this should be joining with `AND`, right? You want to match every entered word in either the URL or the title, not any of them.